ci-operator/step-registry/upi/conf/vsphere/dns/upi-conf-vsphere-dns-commands.sh

```diff
@@ -0,0 +1,138 @@
+            pip install --user awscli
+          else
+            echo "$(date -u --rfc-3339=seconds) - No pip available exiting..."
+            exit 1
+          fi
+      fi
+  fi
+
+  # Load array created in setup-vips:
+  # 0: API
+  # 1: Ingress
+  declare -a vips
+  mapfile -t vips < "${SHARED_DIR}"/vips.txt
+
+  hosted_zone_id="$(aws route53 list-hosted-zones-by-name \
+              --dns-name "${base_domain}" \
+              --query "HostedZones[? Config.PrivateZone != \`true\` && Name == \`${base_domain}.\`].Id" \
+              --output text)"
+  echo "${hosted_zone_id}" > "${SHARED_DIR}/hosted-zone.txt"
+
+  if [ "${JOB_NAME_SAFE}" = "launch" ]; then
+    # setup DNS records for clusterbot to point to the IBM VIP
+    api_dns_target='"TTL": 60,
+      "ResourceRecords": [{"Value": "'${vips[0]}'"}, {"Value": "169.48.190.20"}]'
+    apps_dns_target='"TTL": 60,
+      "ResourceRecords": [{"Value": "169.48.190.20"}]'
+  else
+    # Configure DNS direct to respective VIP
+    api_dns_target='"TTL": 60,
+          "ResourceRecords": [{"Value": "'${vips[0]}'"}]'
+    apps_dns_target='"TTL": 60,
+          "ResourceRecords": [{"Value": "'${vips[1]}'"}]'
+  fi
+
+  # api-int record is needed just for Windows nodes
+  # TODO: Remove the api-int entry in future
+  echo "Creating DNS records..."
+  cat > "${SHARED_DIR}"/dns-create.json <<EOF
+{
+"Comment": "Create public OpenShift DNS records for VSphere IPI CI install",
+"Changes": [{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "api.$cluster_domain.",
+      "Type": "A",
+      $api_dns_target
+      }
+    },{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "api-int.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${vips[0]}"}]
+      }
+    },{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "*.apps.$cluster_domain.",
+      "Type": "A",
+      $apps_dns_target
+      }
+}]}
+EOF
+
+  # api-int record is needed for Windows nodes
+  # TODO: Remove the api-int entry in future
+  echo "Creating batch file to destroy DNS records"
+
+  cat > "${SHARED_DIR}"/dns-delete.json <<EOF
+{
+"Comment": "Delete public OpenShift DNS records for VSphere IPI CI install",
+"Changes": [{
+    "Action": "DELETE",
+    "ResourceRecordSet": {
+      "Name": "api.$cluster_domain.",
+      "Type": "A",
+      $api_dns_target
+      }
+    },{
+    "Action": "DELETE",
+    "ResourceRecordSet": {
+      "Name": "api-int.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${vips[0]}"}]
+      }
+    },{
+    "Action": "DELETE",
+    "ResourceRecordSet": {
+      "Name": "*.apps.$cluster_domain.",
+      "Type": "A",
+      $apps_dns_target
+      }
+}]}
+EOF
+
+  id=$(aws route53 change-resource-record-sets --hosted-zone-id "$hosted_zone_id" --change-batch file:///"${SHARED_DIR}"/dns-create.json --query '"ChangeInfo"."Id"' --output text)
```

Comment:
Consider handling creation errors.

Reply:
I'll look into what errors can happen.  I may request pushing this off if it takes to long so that we can use these changes to test / verify the work being done in the installer repo which is ready for testing / validating.

Reply:
Talked with @jcpowermac and it should error out if an error occurs during the call.  We'll test this a bit more later since its also used in IPI DNS setting and we'll make sure they all perform the same.